File: ai-video-studio/apps/api/src/services/shorts.ts
import { execSync } from "child_process";
import fs from "fs";

export function generateShorts(input: string) {
  fs.mkdirSync("media/video/shorts", { recursive: true });

  // 30s short
  execSync(`
    ffmpeg -y -i ${input} \
    -vf "crop=ih*9/16:ih,scale=1080:1920" \
    -t 30 \
    media/video/shorts/short_30s.mp4
  `);

  // 15s short
  execSync(`
    ffmpeg -y -i ${input} \
    -vf "crop=ih*9/16:ih,scale=1080:1920" \
    -t 15 \
    media/video/shorts/short_15s.mp4
  `);
}

File: ai-video-studio/ffmpeg/shorts.py
import subprocess

def create_short(start, duration, index):
    subprocess.run([
        "ffmpeg",
        "-i", "media/output/final.mp4",
        "-ss", start,
        "-t", duration,
        "-vf", "crop=ih*9/16:ih,scale=1080:1920",
        f"media/video/shorts/short_{index}.mp4"
    ])

create_short("00:00:10", "00:00:30", 1)
create_short("00:02:00", "00:00:30", 2)
create_short("00:05:00", "00:00:30", 3)
